corehq/apps/ota/utils.py

```diff
@@ -165,3 +168,21 @@ def get_restore_user(domain, couch_user, as_user):
         restore_user = couch_user.to_ota_restore_user(domain)
 
     return restore_user
+
+
+def handle_401_response(f):
+    def _inner(request, domain, *args, **kwargs):
+        auth_type = determine_authtype_from_request(request)
+        response = f(request, domain, *args, **kwargs)
```

Comment:
do you need to get the `auth_type` before calling the inner view for some reason? if not, it might be easier to only do it when you need to (`response.status_code == 401`)